src/MainWindow.cpp

```diff
@@ -4562,6 +4578,52 @@ void MainWindow::setupViews()
     showHideViewColumnsForAccountType(currentAccountType);
 }
 
+bool MainWindow::getShowNoteThumbnails() const {
+    QVariant showThumbnails = getApplicationSetting(
+        *m_pAccount, QUENTIER_UI_SETTINGS, LOOK_AND_FEEL_SETTINGS_GROUP_NAME,
+        SHOW_NOTE_THUMBNAILS_SETTINGS_KEY,
+        QVariant::fromValue(DEFAULT_SHOW_NOTE_THUMBNAILS));
+    return showThumbnails.toBool();
+}
+
+QSet<QString> MainWindow::getHideNoteThumbnailsFor() const
+{
+    QVariant hideThumbnailsFor = getApplicationSetting(
+        *m_pAccount, QUENTIER_UI_SETTINGS, LOOK_AND_FEEL_SETTINGS_GROUP_NAME,
+        HIDE_NOTE_THUMBNAILS_FOR_SETTINGS_KEY,
+        QStringLiteral(""));
+
+    // there is QSet::fromList but could nt figure out how to use it with QStringList
+    QSet<QString> hideThumbnailsForSet;
+    const QStringList & values = hideThumbnailsFor.toStringList();
+    foreach (const QString & value, values) {
+        hideThumbnailsForSet.insert(value);
+    }
+    return hideThumbnailsForSet;
+}
+
+void MainWindow::toggleShowNoteThumbnails() const
+{
+    bool newValue = !getShowNoteThumbnails();
+    setApplicationSetting(*m_pAccount, QUENTIER_UI_SETTINGS, LOOK_AND_FEEL_SETTINGS_GROUP_NAME,
+                          SHOW_NOTE_THUMBNAILS_SETTINGS_KEY, QVariant::fromValue(newValue));
+}
+
+void MainWindow::toggleHideNoteThumbnailFor(QString noteLocalUid) const
+{
+    QSet<QString> hideThumbnailsForSet = getHideNoteThumbnailsFor();
+    if (hideThumbnailsForSet.contains(noteLocalUid)) {
+        hideThumbnailsForSet.remove(noteLocalUid);
+    } else {
```

Comment:
Please move this else statement onto the next line to comply with the coding style.